source/s3_client.c

```diff
@@ -305,6 +305,10 @@ struct aws_s3_client *aws_s3_client_new(
         *((size_t *)&client->max_part_size) = (size_t)s_default_max_part_size;
     }
 
+    if (client_config->multipart_upload_threshold != 0) {
+        *((uint64_t *)&client->multipart_upload_threshold) = client_config->multipart_upload_threshold;
```

Comment:
i know we are doing it in other cases, but do we really need all the casting magic here?